### ISSUE TITLE
Doubts about the discriminator gradient direction in WGAN

### ISSUE THREAD

**Issue by fankaisheng** [Zeleni9/pytorch-wgan#43]
In WGAN，the task of discriminator is to predict the Wasserstein-Distance of real_data and fake_data. In the paper, the formula is maximizing Ex~p(real)[f(x)] - Ex~p(fake)[f(x)，in the code it should be errD = err_fake - err_real and calculate the gradient descent parameter update for errD. But I don't see this in the code， just err_real.backward() and err_fake.backward()

**Comment by cjfghk5697** [comment]
@fankaisheng I agree your opinion. WGAN formula exist in the code. But Wasserstein D have not use in the code. It use for the log, not for update.

**Comment by R-N** [comment]
According to [this](https://github.com/jalola/improved-wgan-pytorch/issues/9), it's the same. Since loss.backward(mone) means (-loss).backward(), calling both real.backward(mone) and fake.backward(one) is the same as calling (fake-real).backward().
